docs/curriculum-file-structure.md

```diff
@@ -0,0 +1,78 @@
+# Curriculum File Structure
+
+Our core instructional content is located within the conveniently named `curriculum` directory. This page will break down how these files are organized.
+
+## Terminology
+
+There are a few terms we use when discussing our curriculum content.
+
+- `superBlock` : A superblock is the top level collection of challenges. Each superblock corresponds to a certification in the curriculum (i.e. Responsive Web Design).
+- `block` : A block is a section within a superblock. A block corresponds to a group of challenges in a given certification (i.e. Basic HTML and HTML5)
+- `challenge` : A challenge is a single lesson within the curriculum (i.e. Say Hello to HTML Elements)
+
+## File Tree
+
+Using those terms, here is how the file structure would be defined:
+
+```md
+
+curriculum/
+├─ _meta/
+│  ├─ {block}/
+│  │  ├─ meta.json
+├─ {language}/
+│  ├─ {superBlock}/
+│  │  ├─ {block}/
+│  │  │  ├─ {challenge}.md
+```
+
+## The `_meta` directory
+
+The `_meta` directory is a special directory which contains `.json` files. These files correspond to each block in the curriculum, and are used to determine which superBlock a block belongs to, and the order of the challenges within that block.
+
+## Renaming Files
+
+There may be times when you need to rename a superblock, block, or challenge. This section will outline the steps needed to avoid build errors when doing so.
+
+### Renaming a Superblock
+
+> [!NOTE]
+> When you rename a superblock, the new folder name is used as the path and should be considered the "correct" name. All other values should be updated to reflect that change.
+
+When renaming a superblock, you need to:
+
+1. Rename the superblock folder in the `english` directory.
+2. Rename the superblock folder in *all* language directories.
+3. For each block within that superblock, update the `superBlock` value in the `meta.json` file.
+4. Rename the superblock folder in `client/src/pages/learn`.
+5. Update the `index.md` file in the above folder, changing the `superBlock` value to match the new name.
+6. For each block folder within the above, update the `index.md` to use the correct `superBlock` value.
+7. Update `client/src/redux/index.ts` to use the correct `title` and `certSlug` for the superblock - the `certSlug` being the new folder name.
+8. Update `client/src/resources/cert-and-project.map.ts` in the same manner, updating the path for the cert at the top of the file, and the `title` and `certSlug` for the certificate and the nested `projects` values.
```

Comment:
We should stress there are very rare instances where renaming the `certSlug` is desired - it is a last resort.

Reply:
100%. As we discussed in the back end development issue, each time we do it it adds to the maintenance overhead.

Reply:
Does it not cause an issue if we rename a superblock without changing the `certSlug`??

Reply:
> Does it not cause an issue if we rename a superblock without changing the `certSlug`??

Yes and no. The issue with renaming the `certSlug` is we need redirects for the certificates.

Reply:
Should be resolved with the commit I added